docs/src/main/asciidoc/cdi.adoc

```diff
@@ -0,0 +1,361 @@
+Intializers may accept multiple parameters and don't have to follow the JavaBean naming conventions.
+
+.Initialized and Constructor Injection Example
+[source,java]
+----
+@ApplicationScoped
+public class Translator {
+
+    private final TranslatorHelper helper
+
+    Translator(TranslatorHelper helper) { <1>
+       this.helper = helper;
+    }
+
+    @Inject <2>
+    void setDeps(Dictionary dic, LocalizationService locService) { <3>
+      / ...
+    }
+}
+----
+<1> This is a constructor injection. In fact, this code would not work in regular CDI implementations where a bean with a normal scope must always declare a no-args constructor. In Quarkus we detect this problem and "add" such a constructor directly in the bytecode.
+<2> An initializer method must be annotated with `@Inject`. 
+<3> An initializer may accept multiple parameters - each one is an injection point.
+
+== _Q: You talked about some qualifiers?_
+
+A: https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#qualifiers[Qualifiers, window="_blank"] are annotations that help the container to distinguish beans that implement the same type.
+As we already said a bean is assignable to an injection point if it has all the required qualifiers.
+If you declare no qualifier at an injection point the `@Default` qualifier is assumed.
+
+.Bean With Custom Qualifier Example
+[source,java]
+----
+@Superior <1>
+@ApplicationScoped
+public class SuperiorTranslator extends Translator {
+
+    String translate(String sentence) { 
+      // ...
+    }
+}
+----
+<1> `@Superior` is a https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#defining_qualifier_types[qualifier annotation, window="_blank"].
+
+This bean would be assignable to `@Inject @Superior Translator` and `@Inject @Superior SuperiorTranslator` but not in `@Inject Translator`.
+The reason is that `@Inject Translator` is automatically transformed to `@Inject @Default Translator` during typesafe resolution.
+And since our `SuperiorTranslator` does not declare `@Default` only the original `Translator` bean is assignable.
+
+== _Q: Looks good. What is the bean scope?_
+
+The scope of a bean determines the lifecycle of its instances, i.e. when and where an instance should be created and destroyed.
+
+NOTE: Every bean has exactly one scope.
+
+== _Q: What scopes can I actually use in my Quarkus application?_
+
+A: You can use all the built-in scopes mentioned by the specification except for `javax.enterprise.context.ConversationScoped`.
+
+[options="header",cols="1,1"]
+|===
+|Annotation   |Description   
+//----------------------
+|`@javax.enterprise.context.ApplicationScoped`   | A single bean instance is used for the application and shared among all injection points. The instance is created lazily, i.e. once a method is invoked upon the <<client_proxies, client proxy>>. 
+|`@javax.inject.Singleton`   | Just like `@ApplicationScoped` except that no client proxy is used. The instance is created eagerly.
```

Comment:
The notion of being created eagerly can be misleading. It is eager as in not having to call a method on the bean. However, not as eager as automatically creating it on app startup (which is something we cover with another annotation). I'd probably remove that.

Reply:
_"The instance is created when an injection point that resolves to a `@Singleton` bean is being injected."_? I'd like to keep it because "no client proxy" info is not enough.